test/modules/Instabug.spec.ts

```diff
@@ -236,6 +237,23 @@ describe('Instabug Module', () => {
     await waitForExpect(() => expect(NativeInstabug.reportScreenChange).toBeCalledTimes(2));
   });
 
+  it('setNavigationListener should call the onStateChange on a screen change', async () => {
+    const mockNavigationContainerRef = {
+      current: null,
+      navigate: jest.fn(),
+      reset: jest.fn(),
+      goBack: jest.fn(),
+      dispatch: jest.fn(),
+      getRootState: jest.fn(),
+      canGoBack: jest.fn(),
+      addListener: jest.fn(),
+      removeListener: jest.fn(),
+    } as unknown as NavigationContainerRefWithCurrent<ReactNavigation.RootParamList>;
+
+    Instabug.setNavigationListener(mockNavigationContainerRef);
+    expect(mockNavigationContainerRef.addListener).toBeCalledTimes(1);
```

Comment:
I feel like this test is not really checking that the listener is attached correctly with the correct callback.

I think we should check that it attaches the listener to the `state` event, and that whenever the listener invokes the callback it reports the screen change correctly.